internal/services/machinelearning/machine_learning_registry_resource.go

```diff
@@ -0,0 +1,449 @@
+			}
+
+			param := existing.Model
+
+			if metadata.ResourceData.HasChange("tags") {
+				param.Tags = pointer.To(model.Tags)
+			}
+
+			if metadata.ResourceData.HasChange("identity") {
+				expandedIdentity, err := identity.ExpandLegacySystemAndUserAssignedMapFromModel(model.Identity)
+				if err != nil {
+					return fmt.Errorf("expanding identity: %+v", err)
+				}
+				param.Identity = expandedIdentity
+			}
+
+			if metadata.ResourceData.HasChange("public_network_access_enabled") {
+				param.Properties.PublicNetworkAccess = pointer.To("Disabled")
+				if model.PublicNetworkAccessEnabled {
+					param.Properties.PublicNetworkAccess = pointer.To("Enabled")
+				}
+			}
+
+			if metadata.ResourceData.HasChanges("main_region", "replication_region") {
+				regions := make([]registrymanagement.RegistryRegionArmDetails, 0)
+
+				if len(model.MainRegion) > 0 {
+					regions = append(regions, expandRegistryRegionDetail(model.MainRegion[0]))
+				}
+
+				for _, region := range model.ReplicationRegion {
+					regions = append(regions, expandRegistryRegionDetail(region))
+				}
+
+				param.Properties.RegionDetails = &regions
+			}
+
+			response, err := client.RegistriesCreateOrUpdate(ctx, id, *param)
+			if err != nil && (response.HttpResponse == nil || response.HttpResponse.StatusCode != 202) {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+			if response.HttpResponse != nil {
+				pollerType, err := custompollers.NewMachineLearningRegistryPoller(client, response.HttpResponse)
+				if err != nil {
+					return fmt.Errorf("creating poller: %+v", err)
+				}
+				if pollerType == nil {
+					return fmt.Errorf("no poller created for updating %s", id)
+				}
+				poller := pollers.NewPoller(pollerType, 10*time.Second, pollers.DefaultNumberOfDroppedConnectionsToAllow)
+				if err := poller.PollUntilDone(ctx); err != nil {
+					return fmt.Errorf("polling update of %s: %+v", id, err)
+				}
+			}
+
+			metadata.SetID(id)
+			return nil
+		},
+	}
+}
+
+func (r MachineLearningRegistry) Delete() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.MachineLearning.RegistryManagement
+
+			id, err := registrymanagement.ParseRegistryID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			if err := client.RegistriesDeleteThenPoll(ctx, *id); err != nil {
+				return fmt.Errorf("deleting %s: %+v", *id, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func expandRegistryRegionDetail(input ReplicationRegion) registrymanagement.RegistryRegionArmDetails {
+	return registrymanagement.RegistryRegionArmDetails{
+		Location: pointer.To(input.Location),
+		AcrDetails: &[]registrymanagement.AcrDetails{
+			{
+				SystemCreatedAcrAccount: &registrymanagement.SystemCreatedAcrAccount{
+					AcrAccountSku: pointer.To(string(registrymanagement.SkuTierPremium)),
```

Comment:
Is this not configurable? If not, does it _need_ to be hard coded here, or is there a service default we can/should rely on in case it changes?